src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -304,6 +306,79 @@ private static void UpdateToolObjectToolComponentProperties(JObject tool)
             }
         }
 
+        private static void UpdateAllExternalPropertyFilePropertyTypes(JObject run)
```

Comment:
>UpdateAllExternalPropertyFilePropertyTypes [](start = 28, length = 42)

i think all this code could be replaced by retrieving the run.externalPropertyFiles, then use the helper method to globally rename things.

RecursivePropertyRename(externalPropertyFiles, universallyRenamedMembers);